cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -360,12 +403,12 @@ def update_kubelet_status():
         daemon = 'snap.{}.daemon'.format(service)
         if not _systemctl_is_active(daemon):
             failing_services.append(service)
-
-    if len(failing_services) == 0:
-        hookenv.status_set('active', 'Kubernetes worker running.')
-    else:
+    if failing_services:
```

Comment:
Out of scope for this PR, but since you touched this bit too, something I hit yesterday was the charm reporting status OK when none of the system pods were running.  Might be worth thinking about checking for those as well, or some other minimal health check.